### ISSUE TITLE
warn: Non-local 'impl' definition with #[derive(H5Type)] on nightly

### ISSUE THREAD

**Issue by bramhut** [metno/hdf5-rust#23]
I'm using the macro derive(H5Type) for a simple struct defined as follows:

```rust
#[derive(H5Type, Clone, Copy)]
#[repr(C)]
pub struct Sample(pub [u8; 3]);
```

Everything works as expected, however I am receiving a warning when building using the latest nightly:

```
warning: non-local `impl` definition, `impl` blocks should be written at the same level as their item
  --> src/main.rs:23:10
   |
23 | #[derive(H5Type, Clone, Copy)]
   |          ^-----
   |          |
   |          `H5Type` is not local
   |          move the `impl` block outside of this constant `_IMPL_H5TYPE_FOR_Sample`
24 | #[repr(C)]
25 | pub struct Sample(pub [u8; 3]);
   |            ------ `Sample` is not local
   |
   = note: the derive macro `H5Type` defines the non-local `impl`, and may need to be changed
   = note: the derive macro `H5Type` may come from an old version of the `hdf5_metno_derive` crate, try updating your dependency with `cargo update -p hdf5_metno_derive`
   = note: an `impl` is never scoped, even when it is nested inside an item, as it may impact type checking outside of that item, which can be the case if neither the trait or the self type are at the same nesting level as the `impl`
   = note: items in an anonymous const item (`const _: () = { ... }`) are treated as in the same scope as the anonymous const's declaration for the purpose of this lint
   = note: this lint may become deny-by-default in the edition 2024 and higher, see the tracking issue <https://github.com/rust-lang/rust/issues/120363>
   = note: `#[warn(non_local_definitions)]` on by default
   = note: this warning originates in the derive macro `H5Type` (in Nightly builds, run with -Z macro-backtrace for more info)
```
